File: frontend/src/features/acquisitions/lib/permissions.ts
export function canAddAcquisitions(permissions: string[] | undefined): boolean {
  if (!permissions) return false
  return permissions.includes('*') || permissions.includes('acquisitions.add_landacquisition')
}

export function canEditAcquisitions(permissions: string[] | undefined): boolean {
  if (!permissions) return false
  return permissions.includes('*') || permissions.includes('acquisitions.change_landacquisition')
}

export function canManageAcquisitionWorkflow(permissions: string[] | undefined): boolean {
  if (!permissions) return false
  return permissions.includes('*') || permissions.includes('acquisitions.approve_landacquisition')
}

File: backend/apps/acquisitions/permissions.py
from rest_framework.permissions import BasePermission


class CanManageAcquisitionWorkflow(BasePermission):
    """Gates the approve/mark_purchased/cancel/spawn_project actions."""

    def has_permission(self, request, view):
        return bool(request.user and request.user.has_perm('acquisitions.approve_landacquisition'))
